sql/src/main/java/io/crate/analyze/SetStatementAnalyzer.java

```diff
@@ -22,137 +22,64 @@
 package io.crate.analyze;
 
 import com.google.common.collect.ImmutableList;
-import com.google.common.collect.Sets;
-import io.crate.analyze.expressions.ExpressionToStringVisitor;
-import io.crate.data.Row;
-import io.crate.metadata.settings.CrateSettings;
+import io.crate.analyze.expressions.ExpressionAnalysisContext;
+import io.crate.analyze.expressions.ExpressionAnalyzer;
+import io.crate.analyze.relations.FieldProvider;
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.CoordinatorTxnCtx;
+import io.crate.metadata.Functions;
 import io.crate.sql.tree.Assignment;
-import io.crate.sql.tree.AstVisitor;
 import io.crate.sql.tree.Expression;
-import io.crate.sql.tree.Literal;
-import io.crate.sql.tree.Node;
-import io.crate.sql.tree.ObjectLiteral;
-import io.crate.sql.tree.ParameterExpression;
-import io.crate.sql.tree.ResetStatement;
 import io.crate.sql.tree.SetStatement;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
 
-import java.util.ArrayList;
-import java.util.Collection;
 import java.util.HashMap;
 import java.util.List;
-import java.util.Locale;
 import java.util.Map;
-import java.util.Set;
 
 class SetStatementAnalyzer {
 
-    private static final Logger LOGGER = LogManager.getLogger(SetStatementAnalyzer.class);
+    private final Functions functions;
 
-    public static AnalyzedStatement analyze(SetStatement node) {
+    public SetStatementAnalyzer(Functions functions) {
+        this.functions = functions;
+    }
+
+    public AnalyzedStatement analyze(SetStatement<Expression> node,
+                                     ParamTypeHints typeHints,
+                                     CoordinatorTxnCtx txnCtx) {
 
         boolean isPersistent = node.settingType().equals(SetStatement.SettingType.PERSISTENT);
-        Map<String, List<Expression>> settings = new HashMap<>();
-        Assignment<Expression> assignment;
+        Map<Symbol, List<Symbol>> settings = new HashMap<>();
+
+        var exprAnalyzer = new ExpressionAnalyzer(
+            functions,
+            txnCtx,
+            typeHints,
+            FieldProvider.UNSUPPORTED,
+            null
+        );
+
+        SetStatement<Symbol> statement = node.map(x -> exprAnalyzer.convert(x, new ExpressionAnalysisContext()));
 
         switch (node.scope()) {
```

Comment:
Is it possible to remove this distinction here and only do the `Expression -> Symbol` mapping here?

Currently we switch on scope here and in the planner again. I think either we should try to treat `SetStatement` more generically here, or go the other way around and make concrete statements already in the Parser and treat each statement type completely specific all the way afterwards.

Reply:
Good point, i will give it a try to treat it generically.